pallets/creator-staking/src/functions.rs

```diff
@@ -0,0 +1,462 @@
+        ensure!(staked_value > Zero::zero(), Error::<T>::NotStakedCreator);
+
+        // Calculate the value which will be unstaked.
+        let remaining = staked_value.saturating_sub(desired_amount);
+
+        // If remaining amount is less than minimum staking amount, unstake the entire amount.
+        let amount_to_unstake = if remaining < T::MinimumStake::get() {
+            stake_info.backers_count = stake_info.backers_count.saturating_sub(1);
+            staked_value
+        } else {
+            desired_amount
+        };
+
+        // Sanity check
+        ensure!(amount_to_unstake > Zero::zero(), Error::<T>::CannotUnstakeZero);
+
+        stake_info.total = stake_info.total.saturating_sub(amount_to_unstake);
+
+        backer_info
+            .unstake(current_era, amount_to_unstake)
+            .map_err(|_| Error::<T>::CannotChangeStakeInPastEra)?;
+
+        Self::ensure_max_era_stake_items_not_exceeded(backer_info)?;
+
+        Ok(amount_to_unstake)
+    }
+
+    /// Update the locks for a backer. This will also update the stash lock.
+    /// This lock will lock the entire funds except paying for further transactions.
+    pub(super) fn update_backer_locks(backer: &T::AccountId, backer_locks: BackerLocksOf<T>) {
+        if backer_locks.is_empty() {
+            BackerLocksByAccount::<T>::remove(backer);
+            T::Currency::remove_lock(STAKING_ID, backer);
+        } else {
+            T::Currency::set_lock(STAKING_ID, backer, backer_locks.total_locked, WithdrawReasons::all());
+            BackerLocksByAccount::<T>::insert(backer, backer_locks);
+        }
+    }
+
+    /// Update the backer info for the `(backer, creator_id)` pairing.
+    /// If backer_info is empty, remove it from the DB. Otherwise, store it.
+    pub(super) fn update_backer_info(
+        backer: &T::AccountId,
+        creator_id: CreatorId,
+        backer_info: StakesInfoOf<T>,
+    ) {
+        if backer_info.is_empty() {
+            GeneralBackerInfo::<T>::remove(backer, creator_id)
+        } else {
+            GeneralBackerInfo::<T>::insert(backer, creator_id, backer_info)
+        }
+    }
+
+    /// Calculate the reward distribution between a creator and all their staking participants.
+    ///
+    /// Returns (creator's reward, backers' combined reward)
+    pub(crate) fn distributed_rewards_between_creator_and_backers(
+        creator_info: &CreatorStakeInfo<BalanceOf<T>>,
+        era_info: &EraInfo<BalanceOf<T>>,
+    ) -> (BalanceOf<T>, BalanceOf<T>) {
+        let creator_proportional_stake =
+            Perbill::from_rational(creator_info.total, era_info.staked);
+
+        let creator_reward_share = creator_proportional_stake * era_info.rewards.creators;
+        let combined_backers_reward_share = creator_proportional_stake * era_info.rewards.backers;
+
+        (creator_reward_share, combined_backers_reward_share)
+    }
+
+    /// This utility function converts the specified in a `Config` PalletId into an account ID.
```

Comment:
```suggestion
    /// This utility function converts the PalletId specified in `Config` into an account ID.
```
Config is a specific thing? Or should it be "a `config`"?